x/mongo/driver/topology/topology_test.go

```diff
@@ -693,6 +693,217 @@ func TestTopologyConstruction(t *testing.T) {
 	})
 }
 
+type mockLogSink struct {
+	msgs []string
+}
+
+func (s *mockLogSink) Info(_ int, msg string, _ ...interface{}) {
+	s.msgs = append(s.msgs, msg)
+}
+func (*mockLogSink) Error(error, string, ...interface{}) {
+	// Do nothing.
+}
+
+// Note: SRV connection strings are intentionally untested, since initial
+// lookup responses cannot be easily mocked.
+func TestTopologyConstructionLogging(t *testing.T) {
+	sink := &mockLogSink{}
+	loggerOptions := options.
+		Logger().
+		SetSink(sink).
+		SetComponentLevel(options.LogComponentTopology, options.LogLevelInfo)
+	t.Run("CosmosDB URIs", func(t *testing.T) {
+		testCases := []struct {
+			name string
+			uri  string
+			msgs []string
+		}{
+			{
+				name: "normal",
+				uri:  "mongodb://a.mongo.cosmos.azure.com:19555/",
+				msgs: []string{cosmosDBMsg},
+			},
+			{
+				name: "multiple hosts",
+				uri:  "mongodb://a.mongo.cosmos.azure.com:1955,b.mongo.cosmos.azure.com:19555/",
+				msgs: []string{cosmosDBMsg},
+			},
+			{
+				name: "case-insensitive matching",
+				uri:  "mongodb://a.MONGO.COSMOS.AZURE.COM:19555/",
+				msgs: []string{},
+			},
+			{
+				name: "Mixing genuine and nongenuine hosts (unlikely in practice)",
+				uri:  "mongodb://a.example.com:27017,b.mongo.cosmos.azure.com:19555/",
+				msgs: []string{cosmosDBMsg},
+			},
+		}
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				defer func() {
+					sink.msgs = []string{}
+				}()
+				cfg, err := NewConfig(options.Client().ApplyURI(tc.uri).SetLoggerOptions(loggerOptions), nil)
+				require.Nil(t, err, "error constructing topology config: %v", err)
+
+				topo, err := New(cfg)
+				require.Nil(t, err, "topology.New error: %v", err)
+
+				err = topo.Connect()
+				require.Nil(t, err, "Connect error: %v", err)
+
+				require.ElementsMatch(t, tc.msgs, sink.msgs, "expected messages to be %v, got %v", tc.msgs, sink.msgs)
```

Comment:
Should we use an assertion here? `require` will stop testing if it fails. This applies to the other subtests as well.